### PR TITLE
[Quest]WHM AF1 Messenger From Beyond

### PR DESCRIPTION
**Pull request by hooksta4**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Converts WHM AF1 to IF. 
- Removes logic from NPC files.
- Adds a dialog when spawning NM.
- [Capture](https://www.youtube.com/watch?v=PTgMbnjW1yg)

## Steps to test these changes
Be on whm 40+ Speak with Narchera
Pop and Kill NM 
Return Tavnazia Pass to Narchera

### CODE REVIEW

File: scripts/quests/sandoria/WHM_AF1_Messenger_From_Beyond.lua [hunk 1]
```diff
@@ -0,0 +1,92 @@
+-----------------------------------
+-- Messenger from Beyond
+-- !addquest: 0 87
+-- Narcheral: !pos 128 -11 128 231
+-- qm2      : !pos -714 -9 68 103
+-----------------------------------
+local valkID = zones[xi.zone.VALKURM_DUNES]
```
Comment (claywar) [inline scripts/quests/sandoria/WHM_AF1_Messenger_From_Beyond.lua]: Lets not abbreviate more than necessary, `valkurmID`

File: scripts/quests/sandoria/WHM_AF1_Messenger_From_Beyond.lua [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            ['qm2'] =
+            {
+                onTrigger = function(player, npc)
+                    if
+                        not player:hasItem(xi.item.TAVNAZIA_PASS) and
+                        npcUtil.popFromQM(player, npc, valkID.mob.MARCHELUTE, { claim = true, hide = 0 })
+                    then
+                        return quest:messageSpecial(valkID.text.FOUL_PRESENSE)
+                    else
+                        return quest:messageSpecial(valkID.text.NOTHING_OUT_OF_ORDINARY)
```
Comment (claywar) [inline scripts/quests/sandoria/WHM_AF1_Messenger_From_Beyond.lua]: This should be in DefaultActions

File: scripts/quests/sandoria/WHM_AF1_Messenger_From_Beyond.lua [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            ['Narcheral'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, xi.item.TAVNAZIA_PASS) then
+                        return quest:progressEvent(690) -- Finish quest.
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    return quest:event(688)
```
Comment (claywar) [inline scripts/quests/sandoria/WHM_AF1_Messenger_From_Beyond.lua]: This is not relevant to this quest, and is either a default action or should remain in his script

**Review by claywar — CHANGES_REQUESTED** [review]
Overall fairly reasonable, but removing the default and attaching it to a quest is not acceptable.  This must be changed.

File: scripts/quests/sandoria/WHM_AF1_Messenger_From_Beyond.lua (hunk 3 above)
Comment (hooksta4) [reply scripts/quests/sandoria/WHM_AF1_Messenger_From_Beyond.lua]: made changes to all.. Thanks!

**Pull request closed without merging**